managed/services/telemetry/extension.go

```diff
@@ -0,0 +1,12 @@
+package telemetry
+
+import (
+	"context"
+
+	pmmv1 "github.com/percona-platform/saas/gen/telemetry/events/pmm"
+)
+
+// Extension provides dynamic extension point for Telemetry.
+type Extension interface {
+	FetchMetrics(ctx context.Context, report *Config) ([]*pmmv1.ServerMetric_Metric, error)
+}
```

Comment:
do we really need a separate interface for extensions? it looks similar to `DataSource`

Reply:
btw, why do we use pointer for config here and value in `DataSource`

Reply:
They are similar, but the concept if different. DataSources are predefined in the architecture document.
Idea with extensions is more broad, in the future, if there will be a use case we can have extensions that modify configuration. Or other aspects of telemetry processing.
In future, it can be used as pre/post processor together with DataSource. That is why I didn't want to mix these concepts in the first place.